lara-typescript/src/interactive-api-client/hooks.ts

```diff
@@ -0,0 +1,86 @@
+import { useEffect, useState } from "react";
+import { IInitInteractive } from "./types";
+import * as client from "./api";
+
+export const useInteractiveState = <InteractiveState>() => {
+  const [ interactiveState, setInteractiveState ] = useState<InteractiveState | null>(client.getInteractiveState<InteractiveState>());
+
+  useEffect(() => {
+    // Setup client event listeners. They will ensure that another instance of this hook (or anything else
+    // using client directly) makes changes to interactive state, this hook will receive these changes.
+    const handleStateUpdate = (newState: InteractiveState) => {
+      setInteractiveState(newState);
+    };
+    client.addInteractiveStateListener<InteractiveState>(handleStateUpdate);
+    return () => {
+      client.removeInteractiveStateListener<InteractiveState>(handleStateUpdate);
+    };
+  }, []);
+
+  const handleSetInteractiveState = (state: InteractiveState) => {
+    setInteractiveState(state);
+    client.setInteractiveState<InteractiveState>(state);
+  };
+
+  return { interactiveState, setInteractiveState: handleSetInteractiveState };
```

Comment:
Maybe it would be better to follow React's useState convention and return an array here instead of an object. That way the calling code can decide on its own variable and function names. But I guess unlike useState it doesn't make sense to use this hook multiple times in the same component.

Reply:
Right, I think array is mostly used to let developers use the same hook multiple times. It's not the case here. I think object works a bit better with types and it's easier to extend in the future. But I can change it if you think it's cleaner to follow `useState` convetion.

Reply:
I think it is fine using the object approach, I just wanted to raise the point.